### PR TITLE
Navigation Page

### PR DESCRIPTION
**Pull request by bdrumheller**
This creates the main navigation page used for submitty. 

Checkpoints gradeables and numeric gradeables are no longer on the same view and are accessed with an additional _GET parameter 'g_id'.

The old navigation header on the TA grading side has been removed.

This does not have access control for navigating pages. The default view is the instructor view.

### CODE REVIEW

**Comment by bmcutler** [comment]
My list of improvements:  (some are my opinion can/should be discussed)
* Remove the black horizontal bars beneath the "Future" "Open" etc section titles
* When a section has nothing in it, don't display the section title either
* Do we have an example assignment that should appear in the "Items being graded section"?
* Put year in the date 8-10-16 (or 8/10/16?)
* add "@" before each time?
* For future Items, button text: "SUBMIT (opens 8/10/16 @11:59)"
* For the grading column, use "GRADING" instead of "GRADES"
* The edit button doesn't work for me.  It says:  "You must be an administrator to access this page."  (did I not merge or reinstall something?)
* For graded items, should say "VIEW GRADES (due 8/10/16)" note: no time
* grading buttons color scheme by grading dates not submission dates.

**Comment by bmcutler** [comment]
Also...  remove "Select assignment:" pulldown menu on main submission page.  Yay!

**Comment by MasterOdin** [comment]
Additionally, does this resolve the issues in #475?

**Comment by bdrumheller** [comment]
I've made all of the formatting changes as suggested above.

**Comment by bdrumheller** [comment]
Once merged the issue #475 can be modified

- [ ] ~~Links to editing gradeables page and grading page not added.~~
- [ ] User group check (instructors, TAs, students)
- [ ] ~~Gradeables not sorted chronologically yet.~~
- [ ]  ~~Shortcuts to open grading for a closed item and releasing grades to be added~~
- [ ]  ~~Date checks might be more efficient.~~
- [ ] Progress bars to be added.
- [ ] ~~More styling.~~

File: site/app/views/NavigationView.php [hunk 1]
```diff
@@ -24,16 +24,21 @@ public function showGradeables($sections_to_list) {
...
         $return.= <<<HTML
 HTML;
-        $title_to_button_type = array("FUTURE" => "btn-default", "OPEN" => "btn-primary" , "CLOSED" => "btn-danger", "ITEMS BEING GRADED" => "btn-default", "GRADED" => 'btn-success');
+        $title_to_button_type_submission = array("FUTURE" => "btn-default", "OPEN" => "btn-primary" , "CLOSED" => "btn-danger", 
+                                                 "ITEMS BEING GRADED" => "btn-default", "GRADED" => 'btn-success');
+        $title_to_button_type_grading = array("FUTURE" => "btn-default", "OPEN" => "btn-default" , "CLOSED" => "btn-default", 
+                                                 "ITEMS BEING GRADED" => "btn-primary", "GRADED" => 'btn-danger');
         $title_to_prefix = array("FUTURE" => "OPEN DATE", "OPEN" => "SUBMIT", "CLOSED" => "CLOSED", "ITEMS BEING GRADED" => "GRADING", "GRADED" => "GRADED");
         foreach($sections_to_list as $title => $gradeable_list){
+            if(count($gradeable_list)==0) continue;
```
Comment (MasterOdin) [inline site/app/views/NavigationView.php]: I hate this paradigm of single line if statements (or just not using braces in general) as it's a bit more error-prone at best.

```
if (count($gradeable_list) == 0) {
    continue;
}
```

File: site/app/controllers/student/SubmissionController.php [hunk 2]
```diff
@@ -65,11 +65,7 @@ private function showHomeworkPage() {
                          'action' => 'display',
                          'gradeable_id' => $gradeable->getId());
             $this->core->getOutput()->addBreadcrumb("<a href='{$this->core->buildUrl($loc)}'>{$gradeable->getName()}</a>");
-            
-            $select = $this->core->getOutput()->renderTemplate(array('submission', 'Homework'), 'gradeableSelect',
-                                                               $gradeable_list,
-                                                               $gradeable->getId());
-    
+
```
Comment (MasterOdin) [inline site/app/controllers/student/SubmissionController.php]: Remove this then from https://github.com/Submitty/Submitty/blob/030f0bdc23e9e2c97246ae57e24d72d4016ff1e5/site/app/controllers/student/SubmissionController.php#L71 and https://github.com/Submitty/Submitty/blob/master/site/app/views/submission/HomeworkView.php#L65 and https://github.com/Submitty/Submitty/blob/master/site/app/views/submission/HomeworkView.php#L70 and delete the function at https://github.com/Submitty/Submitty/blob/master/site/app/views/submission/HomeworkView.php#L33

**Pull request merged**